### ISSUE TITLE
Is it possible to remove existing filters

### ISSUE THREAD

**Issue by o-o00o-o** [MicrosoftDocs/powerbi-docs#209]
As I can't get this working I am not sure but if a report has already set some default filters how can I override this. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 70b360f3-21a7-adc2-896a-b1dfd5d9e384
* Version Independent ID: 15e8c09f-46ad-68f9-75e6-c3d5c2c214e5
* Content: [Add Power BI report parameters using the url - Power BI](https://docs.microsoft.com/en-us/power-bi/service-url-filters)
* Content Source: [powerbi-docs/service-url-filters.md](https://github.com/MicrosoftDocs/powerbi-docs/blob/live/powerbi-docs/service-url-filters.md)
* Service: **powerbi**
* GitHub Login: @mihart
* Microsoft Alias: **mihart**

**Comment by mihart** [comment]
@KayUnkroth ,
Reassigning to you since this is a product question, not a doc bug. Thanks.

**Comment by o-o00o-o** [comment]
I would suggest that this is a concept worth mentioning in the docs somewhere, so is a documentation concern?



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by maggiesMSFT** [comment]
Hi, @o-o00o-o -- I checked with the product team. Filters are applied on top of each other. You can't override/remove filters that are already in the report because the URL filters just get added to them. Hope that helps!

**Comment by o-o00o-o** [comment]
Ok thanks however now I wonder how this works if the end user has changed the filters when viewing the report the last time (and the report has not overridden the default setting so that user level filters are applied from last view)?

If the URL report filters are just applied to whatever the current report would show if the users shows them, then this wouldn't give non-deterministic results in that case making this somewhat unreliable in the majority of our cases?

**Comment by KayUnkroth** [comment]
If you reached a report via a filter URL, you would see the impact of changing other filters immediately. You'd for example see if a report now shows no results simply because the filters don't agree with each other. Let's say you filtered customers to only show Contoso, then Litware is no longer available and adding a filter for Litware leads to empty results. Either change the report URL to remove the Contoso filter or remove the filter in the report. But it is always possible to create a filter expression that leads to an empty result. 
Hope this helps.

**Comment by o-o00o-o** [comment]
Hi Kay, I don’t think that this covers my concern although I am not familiar with Contoso. sorry if I didn’t explain my concern well enough before. Let me try again with an example


  1.  I build a report that has 2 filters, country and city available to the user. I publish this report with no default filter value set in either field.
  2.  User A goes to the report and sets the country to USA and city to be New York. Every time from now on that User A goes to that report, filter USA is set and New York is set. Other users may do similar things with different filter values.
  3.  I publish a dashboard with a tile with clickthrough to a URL filter that sets country = UK only (I want the data to show for all cities)

What happens when User A clicks this link?


  1.  Does their previous filter preference get kept and he get no data as there is no New York in UK (bad = nondeterministic result for users which is very unintuitive. All users might get different results)
  2.  Or URL filters always ignore any user filter preference and always bases the given URL filter on the published report filter (good as this is deterministic)
  3.  Or something else..

If it is A how can I “remove the Contoso filter” as you say (or in this case the Country filter). I thought this was the last response I received saying that this isn’t possible?

Brett

**Comment by KayUnkroth** [comment]
So, if I understand you correctly, you'd need something like an additional query string parameter to define the behavior so that users who want to superimpose the URL filter criteria can continue to do so while others would want to ensure only the URL filter criteria is applied can do so as well. Maybe something called ignoreOtherFilters. Would that give you want you need?

**Comment by o-o00o-o** [comment]
It looks like the current behaviour applies the url filter as if the user has never been to the report before, (even if they have and set other filters, that will show on the next time they go to the report without any url filters)  This is good so the only limitation left is that it means that you can't override the filters that the report author published with. I would say ignorePublishedFillters or resetPublishedFilters might be alternative names to allow resetting all of these back to no filters to have the url filters totally deterministic

**Comment by KayUnkroth** [comment]
Yes, we'll think of an appropriate name when the time comes to address this issue. Don't have an ETA yet, but we do have a work item for this on our backlog now.

**Comment by cdxskier** [comment]
Any updates on when this functionality will be added? @KayUnkroth

**Comment by KayUnkroth** [comment]
No updates. Changing the report URL filter behavior is not on the immediate priority list. Sorry, cdxskier, I don't have any timelines for this.
